rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -42,6 +44,9 @@ public class FederationStorageProviderImpl implements FederationStorageProvider
 
     private Script lastRetiredFederationP2SHScript;
 
+    private Sha256Hash fundTransactionUnsignedHash;
+    private boolean isFundTransactionUnsignedHashSet = false;
```

Comment:
What if you use `fundTransactionUnsignedHash` by checking if `null` or not to decide if you should save the fund transaction, so you don't need a second variable to accomplish this?

Reply:
No. Because it is valid to set the `fundTransactionUnsignedHash` as `null`, and that should be saved in storage.

If we check for `fundTransactionUnsignedHash` to be null, then if there is a `fundTransactionUnsignedHash` in the storage, it will not get reset. We need to set the `fundTransactionUnsignedHash` to null, but we only want to access the storage if and only if someone called `setFundTransactionUnsignedHash`, either they set it to `null` or an actual value.